lib/vocabularies/discriminator/index.ts

```diff
@@ -62,10 +63,24 @@ const def: CodeKeywordDefinition = {
       const topRequired = hasRequired(parentSchema)
       let tagRequired = true
       for (let i = 0; i < oneOf.length; i++) {
-        const sch = oneOf[i]
-        const propSch = sch.properties?.[tagName]
-        if (typeof propSch != "object") {
-          throw new Error(`discriminator: oneOf schemas must have "properties/${tagName}"`)
+        let sch = oneOf[i]
+        let propSch
+        if (sch?.["$ref"] && Object.keys(sch).length === 1) {
+          sch = resolveRef.call(it.self, it.schemaEnv, it.baseId, sch?.["$ref"])
+          if (sch instanceof SchemaEnv) {
+            sch = sch.schema
+          }
+          propSch = sch?.properties?.[tagName]
+          if (typeof propSch != "object") {
+            throw new Error(
+              `discriminator: referenced oneOf schemas must have "properties/${tagName}"`
+            )
+          }
+        } else {
+          propSch = sch.properties?.[tagName]
+          if (typeof propSch != "object") {
+            throw new Error(`discriminator: oneOf schemas must have "properties/${tagName}"`)
+          }
```

Comment:
I would take them out of if/else to simplify - the only way they differ is by the error message, which can be unified to be ``` `discriminator: oneOf subschema (or referenced schema) must have "properties/${tagName}"` ```